### PR TITLE
Add serializiation fallback for older janusgraph drivers

### PR DESCRIPTION
**Pull request by farodin91**
Follow up for #2061 to reduce breaking changes between version 0.5 and 0.6.

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -85,10 +85,9 @@ on Cassandra's JVM configurations, Elasticsearch and the old gremlin-server.sh.
 ##### Serialization of JanusGraph predicates has changed
 
 The serialization of JanusGraph predicates has changed in this version for both 
-GraphSON and Gryo. It is therefore necessary to update both the client and 
-the server to this version in parallel as the server will not be able to 
-deserialize a JanusGraph predicate that was serialized by a client prior 
-to version 0.6.0 once it was updated to version 0.6.0.
+GraphSON and Gryo. The newest version of the JanusGraph Driver requires a JanusGraph 
+Server version of 0.6.0 and above. The server includes a fallback for older driver,
```
Comment (FlorianHockmann) [inline docs/changelog.md:89]: I would formulate the last sentence slightly differently and expand it a bit to better explain how the upgrade can be performed now:

> The server includes a fallback for clients with an older driver to make the upgrade to version 0.6.0 easier. This means that the server can be upgraded first without having to update all clients at the same time. The fallback will however be removed in a future version of JanusGraph so clients should also be upgraded.

**Review by FlorianHockmann — COMMENTED** [review]
Apart from my comment for the changelog, I wonder how we could test this. Given that we would need a client from an earlier version to test this, we can probably only test it manually, right? Did you perform a quick test to verify that GraphSON and Gryo still work with an older driver?

**Comment by farodin91** [comment]
> Apart from my comment for the changelog, I wonder how we could test this. Given that we would need a client from an earlier version to test this, we can probably only test it manually, right? Did you perform a quick test to verify that GraphSON and Gryo still work with an older driver?

I will check to build a test.

**Comment by farodin91** [comment]
I found a pretty simple solution, see https://github.com/JanusGraph/janusgraph/pull/2253/files#diff-397bb0e0cc2bcb59b56de6da870cbfcade8f40dbf212dfdee84c55e9cba73180R118-R169.

File: janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java [hunk 2]
```diff
@@ -115,6 +115,58 @@ public void testJanusGraphPredicatesAsGryo() throws SerializationException {
 
     }
 
+    @Test
+    public void testOldFormatJanusGraphPredicatesAsGryo() throws SerializationException {
+        Graph graph = EmptyGraph.instance();
+        GraphTraversalSource g = graph.traversal();
+
+        // Janus Graph Geo, Text Predicates
```
Comment (FlorianHockmann) [inline janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java:123]: (nitpick) The comment doesn't make much sense as there are no Geo predicates here and only one text predicate.

File: janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java [hunk 3]
```diff
@@ -115,6 +115,58 @@ public void testJanusGraphPredicatesAsGryo() throws SerializationException {
...
+        Bytecode oldBytecode = serializeByteCodeAfterDeserializeAsGryo(g.V().has("name", new P<>(Text.CONTAINS, "test")));
```
Comment (FlorianHockmann) [inline janusgraph-driver/src/test/java/org/janusgraph/graphdb/tinkerpop/JanusGraphIoRegistryTest.java:124]: Nice and easy solution!

**Review by FlorianHockmann — APPROVED** [review]

**Pull request merged**